packages/react-native/Libraries/Text/TextInput/RCTBaseTextInputShadowView.m

```diff
@@ -218,7 +218,22 @@ - (NSAttributedString *)measurableAttributedText
 
 - (CGSize)sizeThatFitsMinimumSize:(CGSize)minimumSize maximumSize:(CGSize)maximumSize
 {
-  NSAttributedString *attributedText = [self measurableAttributedText];
+  NSMutableAttributedString *attributedText = [[self measurableAttributedText] mutableCopy];
+
+  /*
+  * The block below is responsible for setting the exact height of the view in lines
+  * Unfortunatelly, iOS doesn't export any easy way to do it. So we set maximumNumberOfLines
+  * prop and then add random lines at the front. However, they are only used for layout
+  * so they are not visible on the screen.
+  */
+  if (self.exactNumberOfLines) {
+    NSMutableString *newLines = [NSMutableString stringWithCapacity:self.exactNumberOfLines];
+    for (NSUInteger i = 0UL; i < self.exactNumberOfLines; ++i) {
+      [newLines appendString:@"\n"];
+    }
+    [attributedText insertAttributedString:[[NSAttributedString alloc] initWithString:newLines attributes:self.textAttributes.effectiveTextAttributes] atIndex:0];
```

Comment:
Does the AttributedString still have all of the various text props that were set?

Text can have things like, different font size in different parts, etc. I'm not sure the most sane way to handle this prop in combination with various metrics.

E.g. we can have an outer TextInput that customizes font size, then a span on the first character which changes it to something else? Which do we pick (it seems like in the java case, we copy all the spans over the beginning of the text).

Reply:
it actually works only when we use same style for the whole TextInput component. I think it makes sense. Let's imagine someone passes Text component as children to textInput and every line has different size. How should that behave? Which line height should we use for missing lines?